example/src/Client/utils/BlockProcessor.ts

```diff
@@ -147,4 +170,63 @@ export class BlockProcessor {
 
     await addNotesToMerkleTree(notes);
   }
+
+  private async _checkForReorg(block: LightBlock) {
+    // If we're not on block 2 or greater, reorgs are impossible.
+    if (!(block.sequence > 1)) {
+      return false;
+    }
+
+    const prevCachedBlock = await this.blockCache.getBlockBySequence(
+      block.sequence - 1,
+    );
+
+    // If the incoming block's previous block hash matches the previous block's hash,
+    // there is no reorg. Note that Buffer.compare returns 0 if the two buffers are equal.
+    if (block.previousBlockHash.compare(prevCachedBlock.hash) === 0) {
+      return false;
+    }
+
+    // Otherwise, we have to walk back the chain until we find the last valid block.
+    let lastValidBlock: LightBlock | null = null;
```

Comment:
Instead of `Valid` can we call this something like `lastMainChainBlock` or something similar. The forked blocks are valid, but they are just not on the main chain.